### PR TITLE
[RFC][POC] Add full Symfony 3 support

### PR DESCRIPTION
**Pull request by WouterJ**
I feel like releasing a version without Symfony 3 support is not a great move at the moment. So I tried to find out what is needed to support Symfony 3. Turns out it means major breaks for this bundle, including dropping Symfony 2.3 support.

We need to discuss if this is a problem or not. As far as I can see, we don't need such major breaks in other CMF bundles to support Symfony 3 (just some ugly lines of code). I don't think it's a huge problem, because:

 * CMF 1.2 still works perfectly on Symfony 2.3 and is quite stable
 * Symfony 2.3 will reach end of maintenance in May 2016
 * There are no BC breaks between Symfony 2.3 and Symfony 2.7/2.8 (the next LTS releases), so it should be quite a smooth upgrade process (in fact, I don't know why, or even if, people are using 2.3 nowadays)

/cc @dantleech @lsmith77 @dbu

### CODE REVIEW

File: Util/Sf2CompatUtil.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ *
+ * (c) 2011-2015 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\CoreBundle\Util;
+
+final class Sf2CompatUtil
```
Comment (WouterJ) [inline Util/Sf2CompatUtil.php]: this utility is quite similair to the one provided by the SonataCoreBundle. However, I don't like depending on another package for something like this. Including such a utility in each bundle means we can make the class as small as possible, by only including the needs for one bundle.

**Comment by dbu** [comment]
the other option could be to release a version without the 3.0 support, then merge the 3.0 support and release a new minor version just shortly after. but i agree with your opinion - if people do not upgrade to symfony 2.7, they probably also don't upgrade the cmf bundles. and if they want new features, they really should upgrade. we can say that we still do important bugfixes on the 1.2 version until 2.3 is end of life, just in case.

**Comment by lsmith77** [comment]
yeah .. I think so too .. lets just keep Symfony 2.3 support in 1.2 alive .. we really do not have the resources to keep tons of BC layers ..

now the bigger question is then if we then also jump to 2.0?

**Comment by dbu** [comment]
imho we should tag something that works with 2.8 (using deprecated things if needed) asap and then do large refactorings for symfony 3 and bump our version to 2.0. or do we not have any features that warrant a 1.3 release at all?

**Comment by patkar** [comment]
backref: sulu-io/sulu#2106

**Comment by WouterJ** [comment]
This PR is now ready to be merged. Tests are failing because RoutingBundle 2.0 doesn't support Symfony 3 yet. See a passing test with a sf3 supporting version here: https://travis-ci.org/symfony-cmf/core-bundle/builds/127007822

**Comment by ElectricMaxxx** [comment]
Shouldn't we remove the class parameters too?

**Comment by WouterJ** [comment]
Yeah and many other things as well. But this PR purely focuses on Symfony 3 support

**Pull request merged**